internal/common/expand_into_runfiles.bzl

```diff
@@ -67,28 +111,32 @@ def expand_location_into_runfiles(ctx, input, targets = []):
     """
     target = "@%s//%s:%s" % (ctx.workspace_name, "/".join(ctx.build_file_path.split("/")[:-1]), ctx.attr.name)
 
-    # Loop through input an expand all $(location) and $(locations) using _expand_to_mlocation()
+    # Loop through input an expand all predefined source/output path variables
+    # See https://docs.bazel.build/versions/master/be/make-variables.html#predefined_label_variables.
     path = ""
     length = len(input)
     last = 0
     for i in range(length):
-        if (input[i:i + 12] == "$(mlocation ") or (input[i:i + 13] == "$(mlocations "):
+        # Support legacy $(location) and $(locations) expansions which return the runfiles manifest path
+        # in the format `repo/path/to/file`. This expansion is DEPRECATED. See docstring above.
+        # TODO: Change location to behave the same as the built-in $(location) expansion for args of *_binary
+        #       and *_test rules. This would be a BREAKING CHANGE.
+        if input[i:].startswith("$(location ") or input[i:].startswith("$(locations "):
```

Comment:
opt: seems like there's enough conditional logic and string manipulation in this code that we ought to have a starlark unit test for it

Reply:
Yea. I was thinking about that but wasn't top of list. I'll add it today.

Reply:
Done.